src/Core/Components/TreeView/FluentTreeView.razor.cs

```diff
@@ -49,6 +54,18 @@ public partial class FluentTreeView : FluentComponentBase, IDisposable
     [Parameter]
     public EventCallback<FluentTreeItem> OnExpandedChange { get; set; }
 
+    /// <summary>
+    /// Gets or sets the template for rendering tree items.
+    /// </summary>
+    [Parameter]
+    public RenderFragment<ITreeViewItem>? ItemTemplate { get; set; }
+
+    /// <summary>
+    /// Gets or sets whether the tree should use lazy loading when expanding nodes
+    /// </summary>
```

Comment:
Just from the documentation it is not super clear what this parameter does. I'm thinking maybe it will call a user-provided method to get the next nodes or only render the expanded nodes. Though there is already `RenderCollapsedNodes` that should take care of that.

Reply:
It's a `boolean`, so the list of items is already known, but is not rendered (HTML) if the current element is reduced.
We could perhaps add another method to be called when the Expand event is triggered.

Reply:
So what's the difference with `RenderCollapsedNodes` ? 🤔

Reply:
- `RenderCollapsedNodes` is an attribute included in the TreeView Web component. The goal seems similar, but it doesn't work properly with this attribute. probably because it keeps all the data in memory (JavaScript).

- This new `LazyLoadItems` works with Blazor's SignalR rendering and doesn't send sub-elements until they're requested. In the example above, only the first 15 items are actually present in the HTML page, allowing it to be displayed instantly on screen.

I think we should tag `RenderCollapsedNodes` as **Obsolete** and suggest using `LazyLoadItems`.
@vnbaaij What do you think?

Reply:
> LazyLoadItems works with Blazor's SignalR rendering

Does it mean it would not work for standalone wasm?

Reply:
I don't understand

Reply:
> I think we should tag `RenderCollapsedNodes` as **Obsolete** and suggest using `LazyLoadItems`.

I think that makes sense to do so, yes

Reply:
Blazor supports several hosting modes:
- Blazor Server
- Blazor WebAssembly
- Blazor Hybrid

In Blazor WebAssembly, there are no SignalR server. Would the LazyLoadItems feature still work then?

Reply:
Of course, it's Blazor's job to render Blazor Server or Blazor WASM code correctly :-)

Reply:
Got it thanks!

Reply:
> I think we should tag RenderCollapsedNodes as Obsolete and suggest using LazyLoadItems.

Alternatively, if the 2 features have similar goal but different implementations it could make sense not to introduce `LazyLoadItems` but to change `RenderCollapsedNodes` implementation with the new one.

Reply:
Except if some people are already used this feature and don't want to change the unit test, for example, or other stuff.